src/main/java/org/openstreetmap/atlas/utilities/testing/TestAtlasHandler.java

```diff
@@ -320,6 +335,33 @@ private void handle(final PackedAtlasBuilder builder, final Relation... relation
         }
     }
 
+    private void loadFromJosmOsmResource(final Field field, final CoreTestRule rule,
+            final CreationContext context, final String resourcePath)
+    {
+        final String packageName = rule.getClass().getPackage().getName().replaceAll("\\.", "/");
+        final String completeName = String.format("%s/%s", packageName, resourcePath);
+        final ClassResource resource = new ClassResource(completeName);
+        FileSuffix.suffixFor(completeName).ifPresent(suffix ->
+        {
+            if (suffix == FileSuffix.GZIP)
+            {
+                resource.setDecompressor(Decompressor.GZIP);
+            }
+        });
+        try
+        {
+            final StringResource osmFile = new StringResource();
+            new OsmFileParser().update(resource, osmFile);
+            final ByteArrayResource pbfFile = new ByteArrayResource();
+            new OsmFileToPbf().update(osmFile, pbfFile);
+            field.set(rule, new OsmPbfLoader(pbfFile, AtlasLoadingOption.withNoFilter()).read());
+        }
+        catch (IllegalArgumentException | IllegalAccessException e)
+        {
+            throw new CoreException("Error loading from text resource {}", resourcePath, e);
```

Comment:
from OSM resource? I guess it's text, that may be confusing if people think it's a .txt resource.

Reply:
Nice catch too!